src/Bicep.Core/Parsing/ParamsParser.cs

```diff
@@ -74,13 +75,42 @@ public override ProgramSyntax Program()
         private UsingDeclarationSyntax UsingDeclaration()
         {
             var keyword = ExpectKeyword(LanguageConstants.UsingKeyword);
+
+            SyntaxBase expression = reader.Peek().Type switch
+            {
+                TokenType.EndOfFile or
+                TokenType.NewLine or
+                TokenType.NoneKeyword => WithRecovery(
+                    () => new NoneLiteralSyntax(
+                        Expect(
+                            TokenType.NoneKeyword,
+                            e => e.ExpectedKeyword(LanguageConstants.NoneKeyword)
+                        )
+                    ),
+                    GetSuppressionFlag(keyword),
+                    TokenType.NewLine
+                ),
+                TokenType.StringComplete => ThrowIfSkipped(this.InterpolableString, b => b.ExpectedFilePathString()),
+                _ => Skip(reader.Read(), b => b.ExpectedSymbolListOrWildcard()),
+            };
+
+            return new(keyword, expression);
+        }
+
+        private ExtendsDeclarationSyntax ExtendsDeclaration()
+        {
+            var keyword = ExpectKeyword(LanguageConstants.ExtendsKeyword);
             var path = this.WithRecovery(
-                () => ThrowIfSkipped(this.InterpolableString, b => b.ExpectedFilePathString()),
+                () => ThrowIfSkipped(this.InterpolableString, b => b.ExtendsPathHasNotBeenSpecified()),
                 GetSuppressionFlag(keyword),
-                TokenType.Assignment, TokenType.NewLine);
+                TokenType.NewLine);
 
-            return new UsingDeclarationSyntax(keyword, path);
+            if (path is StringSyntax pathSyntax && pathSyntax.SegmentValues[0] == string.Empty)
+            {
+                throw new ExpectedTokenException(reader.Peek(), b => b.ExtendsPathHasNotBeenSpecified());
+            }
```

Comment:
Why is this logic necessary? How do we handle interpolation for `using`? Could we follow a similar approach for `extends`?

Reply:
I removed the unneeded if statement, it was here because at the beginning we think the syntax as `using 'none'`, then decided as `using none`